pkg/patterns/declarative/reconciler.go

```diff
@@ -63,29 +69,44 @@ type DeclarativeObject interface {
 // For mocking
 var kubectl = kubectlcmd.New()
 
-func (r *Reconciler) Init(mgr manager.Manager, prototype DeclarativeObject, opts ...reconcilerOption) error {
+func (r *Reconciler) Init(mgr manager.Manager, prototype DeclarativeObject, opts ...reconcilerOption) (err error) {
 	r.prototype = prototype
 	r.kubectl = kubectl
 
 	r.client = mgr.GetClient()
 	r.config = mgr.GetConfig()
 	r.mgr = mgr
+	globalObjectTracker.mgr = mgr
 
-	if err := r.applyOptions(opts...); err != nil {
-		return err
+	if err = r.applyOptions(opts...); err != nil {
+		return
+	}
+
+	metrics.Registry.MustRegister(reconcileCount, reconcileFailure, managedObjectsRecord)
```

Comment:
Can you make Metrics tracking an optional (default off) feature then update the [example](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/tree/master/docs/addon/walkthrough#using-the-framework-in-the-controller) to turn it on?

This is the pattern we've been following (eg with [WithApplyPrune](https://pkg.go.dev/sigs.k8s.io/kubebuilder-declarative-pattern@v0.0.0-20200522144838-848d48e5b073/pkg/patterns/declarative?tab=doc#WithApplyPrune)). This allows us to keep a stable API/functionality for users while introducing new features.

I expect, not so far in the future, we have a 'Recommended Set of Options' and I would certainly expect this to be apart of that. For now, our recommended set is the ones in the walkthrough.